.github/workflows/release.yaml

```diff
@@ -393,18 +398,27 @@ jobs:
       - linux-cross
       - musllinux
       - musllinux-cross
-    if: "startsWith(github.ref, 'refs/tags/')"
+    # If you don't set an input it's a practice run
+    if: "${{ inputs.tag }} != ''"
     environment:
       name: release
     permissions:
       # For pypi trusted publishing
       id-token: write
     steps:
+      - name: Consistency check
+        run: |
+          version=$(grep "version = " pyproject.toml | sed -e 's/version = "\(.*\)"/\1/g')
+          echo "${{ inputs.tag }}"
+          echo "${version}"
+          if [ "${{ inputs.tag }}" != "${version}" ]; then
+            exit 1
```

Comment:
Can we exit with an error message that tells the user what went wrong?

Reply:
I'm printing the versions so you can compare, but otherwise it doesn't look like bash has an equivalent to `bail!("tag mismatch: {} {}", tag, version)` (https://stackoverflow.com/a/24597890/3549270)

Reply:
I think an echo would be useful, mentioning that some files still reference the old version (what happens if one of our rust dependencies use the same version number?)